blocks/sign-up/sign-up.js

```diff
@@ -0,0 +1,148 @@
+ */
+function blockNonNumbers(event) {
+  const inputElement = event.target;
+  inputElement.value = inputElement.value.replace(/[^0-9]/g, '');
+}
+
+function createMobileNumberInput(placeholderText) {
+  const inputElement = createElement('input', 'phonenumber-textbox');
+  inputElement.type = 'text';
+  inputElement.placeholder = placeholderText;
+  inputElement.maxLength = '10';
+  inputElement.pattern = '[0-9]*';
+  inputElement.autocomplete = 'off';
+  inputElement.addEventListener('input', blockNonNumbers);
+  return inputElement;
+}
+
+function createSubmitButton(buttontitle) {
+  const submitButton = createElement('button', 'signupbtn');
+  submitButton.textContent = buttontitle;
+  submitButton.addEventListener('click', handleOpenAccountSubmit);
+  return submitButton;
+}
+function createErrorSpan(errormessage) {
+  const spanElement = createElement('span', 'error-message');
+  spanElement.textContent = errormessage;
+  return spanElement;
+}
+function createTitle(titleHTML) {
+  const col1Div = createElement('div', 'signupsections');
+  const titleWrapDiv = createElement('div', 'title-wrap');
+  const h2Element = createElement('h2', '');
+  h2Element.innerHTML = titleHTML.innerHTML;
+  h2Element.classList.add('text-left');
+  titleWrapDiv.appendChild(h2Element);
+  col1Div.appendChild(titleWrapDiv);
+  return col1Div;
+}
+function createSignUpElement(
+  signupString,
+  promotionaltext,
+  placeholderText,
+  buttontitle,
+  errormessage,
+) {
+  const signupFormDiv = createElement('div', 'signupsections');
+
+  const formGroupDiv = createElement('div', 'signup-form-group');
+  formGroupDiv.classList.add('text-center');
+
+  const signupStringElement = createElement('label', '');
+  signupStringElement.innerHTML = signupString.innerHTML;
+
+  const promotionalSpan = createElement('span', 'promotional-text');
+  promotionalSpan.textContent = promotionaltext;
+
+  const formFieldsDiv = createElement('div', '');
+  const mobileInput = createMobileNumberInput(placeholderText);
+  const tunrstileContainer = createElement('div', 'turnstile-container');
+  const submitButton = createSubmitButton(buttontitle, errormessage);
+  const errorSpan = createErrorSpan(errormessage);
+
+  formFieldsDiv.appendChild(mobileInput);
+  formFieldsDiv.appendChild(tunrstileContainer);
+  formFieldsDiv.appendChild(submitButton);
+
+  formGroupDiv.appendChild(signupStringElement);
+  formGroupDiv.appendChild(promotionalSpan);
+  formGroupDiv.appendChild(formFieldsDiv);
+  formGroupDiv.appendChild(errorSpan);
+
+  signupFormDiv.appendChild(formGroupDiv);
+  return signupFormDiv;
+}
+/**
+ * Finds an HTML element from a block based on a given key.
+ * @param {HTMLElement} block - The block to search within.
+ * @param {string} key - The key to search for within the block.
+ * @returns {HTMLElement|string} - The found HTML element or an empty string if no match is found.
+ */
+function findHTMLElementFromBock(block, key) {
```

Comment:
I believe this methods is using for fetching elements, we should avoid iteration overall block to find desired element.
Instead use `.getElementById` and `.querySelector` , And I hope it will improve the performance as well.

Reply:
Changed to querySelector as recommended but this will fixed order of properties in the table.